pkg/expr/func.go

```diff
@@ -126,13 +124,17 @@ func (f *eqFunc) call(args0 interface{}, args1 interface{}) bool {
 		}
 		return matchWithWildcards(s0, s1)
 	case []byte:
-		if len(args0.([]byte)) == net.IPv4len || len(args0.([]byte)) == net.IPv6len {
+		return bytes.Equal(s0, args1.([]byte))
+	case net.IP:
+		if len(s0) == net.IPv4len || len(s0) == net.IPv6len {
```

Comment:
this is only needed in the []byte case. we don't need it for cases where we already know we have net.IPs.